SonarQube.TeamBuild.PreProcessor/Interfaces/IAnalyzerProvider.cs

```diff
@@ -19,6 +20,6 @@ public interface IAnalyzerProvider
         /// </summary>
         /// <param name="projectKey">Identifier for the project being analyzed</param>
         /// <returns>The settings required to configure the build for Roslyn analysis</returns>
-        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey);
+        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey, string projectBranch);
```

Comment:
If you mean create a new class called "ProjectId" that holds the key and branch as separate values, yes.
If you mean concatenate the key and branch into a single string, no.
Project Key and Project Branch are both SonarQube concepts and are exposed on their APIs. We don't know how they are used internally by SonarQube so we can't combine them into a single value. All we can do is pass them on when calling SonarQube APIs.
The only piece of our code that needs to know how to the manipulate the key and branch values is our implementation of ISonarQubeServer - i.e. the code that is actually constructing the URL.